src/main/java/TaskList.java

```diff
@@ -0,0 +1,120 @@
+import java.util.ArrayList;
+
+/**
+ * Class to represent a list of tasks
+ */
+public class TaskList {
+
+    ArrayList<Task> tasks = new ArrayList<>();
+    int taskCount = 0;
+
+    /**
+     * Returns current list of tasks
+     * @return current list of tasks
+     */
+    protected ArrayList<Task> getUpdatedTasks() {
+        return tasks;
+    }
+
+    /**
+     * Calls function to get description for specific task in list
+     * @param index of specific task
+     * @return description of specific task
+     */
+    public String getDescription(int index) {
+        return tasks.get(index).getDescription();
+    }
+
+    /**
+     * Calls function to get status icon for specific task in list
+     *
+     * @param index of specific task
+     * @return status icon of specific task
+     */
+    public String getStatusIcon(int index) {
+        return tasks.get(index).getStatusIcon();
+    }
+
+    /**
+     * Calls function to get task icon for specific task in list
+     *
+     * @param index of specific task
+     * @return task icon of specific task
+     */
+    public String getTaskIcon(int index) {
+        return tasks.get(index).getTaskIcon();
+    }
+
+
+    /**
+     * Add new to do to list and return its index
+     *
+     * @param input text entered by user
+     * @return index of new deadline
+     * @throws DukeException
+     */
+    protected int addToDo(String input) throws DukeException {
+        if (input.length() <= ("todo".length() + 1)) {
+            throw new DukeException();
+        }
+        tasks.add(new ToDo(input.substring("todo".length() + 1)));
+        taskCount++;
+        return taskCount - 1;
```

Comment:
Not necessary to have a taskCount counter. You can utilise the ArrayList built in function of .size() to get the total number of task currently in the list. Reduces the line you need to write (removing taskCount++) and less susceptible for error!